### PR TITLE
Python 3: fix format class string/bytes issues

### PR DESCRIPTION
**Pull request by Anthchirp**
Mixed bag of format class changes fixing places where we identify files using some byte signatures and compared them to strings which only works in Python 2.

### CODE REVIEW

**Comment by Anthchirp** [comment]
dxtbx and dials tests pass, no dxtbx 3 regressions

File: format/FormatBruker.py [hunk 1]
```diff
@@ -118,7 +120,7 @@ def _goniometer(self):
     def _detector(self):
         """Return a model for a simple detector"""
 
-        twotheta = self.detectorbase.parameters["TWOTHETA"]
+        # twotheta = self.detectorbase.parameters["TWOTHETA"]
         # At present, ignore non-zero two theta for the dxtbx model
```
Comment (graeme-winter) [inline format/FormatBruker.py]: Would probably help to have `# At present` before ignoring twotheta

File: format/FormatSER.py [hunk 2]
```diff
@@ -32,37 +31,36 @@ def __init__(self, image_file, **kwargs):
...
             return False
 
         # DataTypeID: 0x4122 if elements are 2D arrays
-        if not struct.unpack("<I", tag[6:10])[0] == 16674:
+        if struct.unpack("<I", tag[6:10])[0] != 16674:
             return False
 
         # TagTypeID: 0x4142 or 0x4152
-        if not struct.unpack("<I", tag[10:])[0] in [16706, 16722]:
+        if struct.unpack("<I", tag[10:])[0] not in (16706, 16722):
```
Comment (graeme-winter) [inline format/FormatSER.py]: Since we are testing for hex values, better to use hex integer constants (appreciate unchanged in the pR; campsite rules)

Comment (Anthchirp) [reply format/FormatSER.py]: I didn't even see that when refactoring, so very valid point.

**Pull request merged**